src/main/java/uk/gov/ons/ssdc/supporttool/endpoint/SurveyCasesEndpoint.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import uk.gov.ons.ssdc.supporttool.model.dto.CaseSearchResult;
+import uk.gov.ons.ssdc.supporttool.model.dto.ui.UIRefusalTypeDTO;
+import uk.gov.ons.ssdc.supporttool.model.entity.Survey;
+import uk.gov.ons.ssdc.supporttool.model.entity.UserGroupAuthorisedActivityType;
+import uk.gov.ons.ssdc.supporttool.model.repository.SurveyRepository;
+import uk.gov.ons.ssdc.supporttool.security.UserIdentity;
+import uk.gov.ons.ssdc.supporttool.utility.CaseSearchResultsMapper;
+
+@RestController
+@RequestMapping(value = "/searchInSurvey")
+public class SurveyCasesEndpoint {
+
+  private final SurveyRepository surveyRepository;
+  private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+  private final CaseSearchResultsMapper caseRowMapper;
+  private final UserIdentity userIdentity;
+
+  private static final String searchCasesPartialQuery =
+      "SELECT c.id, c.case_ref, c.sample, c.collection_exercise_id, e.name collex_name";
+  private static final String searchCasesInSurveyPartialQuery =
+      searchCasesPartialQuery
+          + " FROM casev3.cases c, casev3.collection_exercise e WHERE c.collection_exercise_id = e.id"
+          + " AND e.survey_id = :surveyId";
+
+  public SurveyCasesEndpoint(
+      SurveyRepository surveyRepository,
+      NamedParameterJdbcTemplate namedParameterJdbcTemplate,
+      CaseSearchResultsMapper caseRowMapper,
+      UserIdentity userIdentity) {
+    this.surveyRepository = surveyRepository;
+    this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
+    this.caseRowMapper = caseRowMapper;
+    this.userIdentity = userIdentity;
+  }
+
+  @GetMapping(value = "/{surveyId}")
+  @ResponseBody
+  public List<CaseSearchResult> searchCasesBySampleData(
+      @RequestHeader(required = false, value = "x-goog-iap-jwt-assertion") String jwt,
+      @PathVariable(value = "surveyId") UUID surveyId,
+      @RequestParam(value = "searchTerm") String searchTerm,
+      @RequestParam(value = "collexId", required = false) Optional<UUID> collexId,
+      @RequestParam(value = "receipted", required = false) Optional<Boolean> receiptReceived,
+      @RequestParam(value = "invalid", required = false) Optional<Boolean> addressInvalid,
+      @RequestParam(value = "launched", required = false) Optional<Boolean> surveyLaunched,
+      @RequestParam(value = "refusal", required = false)
+          Optional<UIRefusalTypeDTO> refusalReceived) {
+
+    checkSurveySearchCasesPermission(jwt, surveyId);
+
+    searchTerm = '%' + searchTerm + '%';
```

Comment:
I don't think you should mutate a supplied parameter... bad practice.

Reply:
also, using `+` for string concatenation is inefficient. Always prefer `String.format` or `StringBuilder`